x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -124,14 +125,28 @@ private void createFollowerIndex(
             return;
         }
 
-        final Settings.Builder settingsBuilder = Settings.builder()
+        final Settings replicatedRequestSettings = TransportResumeFollowAction.filter(request.getSettings());
+        if (replicatedRequestSettings.isEmpty() == false) {
+            final String message = String.format(
+                Locale.ROOT,
+                "can not put follower index that could override leader settings %s",
+                replicatedRequestSettings
+            );
+            listener.onFailure(new IllegalArgumentException(message));
+            return;
+        }
+
+        final Settings overrideSettings = Settings.builder()
             .put(IndexMetadata.SETTING_INDEX_PROVIDED_NAME, request.getFollowerIndex())
-            .put(CcrSettings.CCR_FOLLOWING_INDEX_SETTING.getKey(), true);
+            .put(CcrSettings.CCR_FOLLOWING_INDEX_SETTING.getKey(), true)
+            .put(request.getSettings())
```

Comment:
This is the crux of the change, where the request settings are propagated to the follower index. The rest of this change is just ceremony around this, parsing the settings from the request, validating that they are not including any settings that must be propagated from the leader index, and tests around this.